src/OrchardCore/OrchardCore.DisplayManagement/Views/ShapeResult.cs

```diff
@@ -107,11 +107,16 @@ private async Task ApplyImplementationAsync(BuildShapeContext context, string di
 
         if (!string.IsNullOrEmpty(groupId))
         {
-            _groupIds = StringValues.Concat(_groupIds, groupId);
+            OnGroup(groupId);
         }
 
         // If the shape's group doesn't match the currently rendered one, return.
-        if (!string.IsNullOrEmpty(context.GroupId) && !_groupIds.Contains(context.GroupId, StringComparer.OrdinalIgnoreCase))
+        if (_groupIds.Count > 0 && !_groupIds.Contains(context.GroupId ?? string.Empty, StringComparer.OrdinalIgnoreCase))
+        {
+            return;
+        }
+
+        if (_groupIds.Count == 0 && !string.IsNullOrEmpty(context.GroupId))
```

Comment:
I think that's what `StringValues.IsNullOrEmpty` does

Reply:
updated